Data/Map/Lazy.hs

```diff
@@ -14,19 +14,63 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- An efficient implementation of ordered maps from keys to values
--- (dictionaries).
 --
--- API of this module is strict in the keys, but lazy in the values.
--- If you need value-strict maps, use "Data.Map.Strict" instead.
--- The 'Map' type itself is shared between the lazy and strict modules,
--- meaning that the same 'Map' value can be passed to functions in
--- both modules (although that is rarely needed).
+-- = Lazy Maps (aka dictionaries)
 --
--- These modules are intended to be imported qualified, to avoid name
--- clashes with Prelude functions, e.g.
+-- The @'Map' k v@ type represents an /ordered/ map indexed by keys of type @k@
+-- containing /lazy/ values of type @v@.
```

Comment:
What is an "ordered map"? Again, the `Map` *type* here is exactly the same as the one in `.Strict`. I would probably change the intro to something like this:

    -- The @'Map' k v@ type represents a finite map (sometimes called a dictionary)
    -- from keys of type @k@ to values of type @v@. A 'Map' is strict in its keys but lazy
    -- in its values.
    --
    -- The functions in "Data.Map.Strict" are careful to force values before installing
    -- them in a 'Map'. This is usually more efficient in cases where laziness is not essential.
    -- The functions in this module do not do so.

If you make the change this way, please remove the first bullet point below.